File: QwStacks.py
class QwStacks():
    def __init__(self):
        self.__stackA = []
        self.__stackB = []

    def atob(self):
        while len(self.__stackA)>0:
            self.__stackB.append(self.__stackA.pop())

    def size(self):
        return len(self.__stackA) + len(self.__stackB)

    def add(self,item):
        self.__stackA.append(item)

    def remove(self, item):
        if len(self.__stackB) == 0:
            self.atob()
        return self.__stackB.pop()

    def peek(self):
        if len(self.__stackB) == 0:
            self.atob()
        return self.__stackB[-1]

    def isEmpty(self):
        return self.size() == 0

File: QwStacks.java
import java.util.Stack;
import java.util.EmptyStackException;

public class QwStacks <T>{
  private Stack <T> stackA, stackB;
  public QwStacks(){
    stackA = new Stack <T>();
    stackB = new Stack <T>();
  }

  public void AtoB(){
    while (!stackA.empty()){
      stackB.push(stackA.pop());
    }
  }

  public void add(T item){
    stackA.push(item);
  }

  public T remove(){
    if (isEmpty()){
      throw new EmptyStackException();
    }
    if(stackB.empty()){
      AtoB();
    }
    return stackB.pop();
  }
  public T peek(){
    if (stackB.empty()){
      AtoB();
    }
    return stackB.peek();
  }

  public boolean isEmpty(){
    return stackA.empty() && stackB.empty();
  }
}
